### ISSUE TITLE
Browser used to post on the discussion forum.

### ISSUE THREAD

**Issue by TimidScript** [OpenUserJs/OpenUserJS.org#764]
Posts on the discussion forum should show the browser used to create it.

**Comment by Martii** [comment]
Probably included as a dependency of #601 ... not sure... will need to wait for @sizzlemctwizzle to pipe in.

**Comment by sizzlemctwizzle** [comment]
I like the idea. But it would impact the UI of #601, whenever that gets
done. If anyone wants to do this, they also need to complete some of the UI
for #601:

* put in disabled edit and deleted buttons on the logged in user's comments

* put a disabled flag icon button on all comments when the user is logged
in (I can handle the extra conditions of when not to show this)

* put in a disabled remove button when the logged in user has the role of
admin or greater (technically *less* in the code)

I'd be much more happy to do the backend changes for #601 if I knew the
frontend UI was just waiting for me. The frontend is really what keeps #601
unresolved since I've never worked with boostrap and wouldn't know where to
start.

Bottom line is that since the changes to the UI for this issue are in
roughly the same places as those for #601, I won't accept a PR for this
issue without the above additional changes. I know that makes me sound like
a dick, but this is a cosmetic change, and if you're going to do it, you
might as well also do something productive to help along a more important
issue.

If @Marti doesn't want to do all this, and you really don't have to man
(you do way more than enough work already), then we should just leave this
until #601 is finished.

**Comment by Martii** [comment]
I'll create this as a milestone dependent for #601 then.

I'm still hunting for a toolbar package but none seems to exist that I can find. That I could use another few sets of eyes looking for on npmjs.com *(hint hint)*.

As far as *bootstrap* goes it's just a bunch of classes that modify things to fit into a theme... **the biggest thing we would need to have determined is placement.** *(need mockups preferred here)* If we were to mimic placement like USO did it would be [here](https://github.com/OpenUserJs/OpenUserJS.org/blob/e946670737da259a179811c4422ec5972e16155d/views/includes/comment.html#L4-L8) for this issue.

Then somewhere around [here](https://github.com/OpenUserJs/OpenUserJS.org/blob/e946670737da259a179811c4422ec5972e16155d/models/comment.js#L13-L27) we would need to start capturing the User-Agent in those routes.

A lot of bug checking has to occur in order to show the right icon... fair use will allow us to store those images on the server just like my Compatibility Matrix icons. We also need to decide on a default to show *(or not to show)*... User-Agent spoofers are always going to wreak havoc with this issue RFE btw.

Also would need to consider if they should be grey-scaled instead of colorized... most of our icons are never colored... which brings up another point... perhaps there is an icon package *(and or font directly)* with **all** the browser icons in them.

I've thought about this... but haven't gotten around to mentioning it and it's repercussions. I am stretched a bit thin lately with all the bug checking but can milestone it to break it up into parts.

**Comment by Martii** [comment]
We could also put it [next to the date](https://github.com/OpenUserJs/OpenUserJS.org/blob/e946670737da259a179811c4422ec5972e16155d/views/includes/comment.html#L16-L20) too... which leads into the "last edited" feature in #601 ... could put it in as original browser agent and last edited agent too... depends on how elaborate we want to do this.

**Comment by Martii** [comment]
Btw I'm currently against actually putting the full User-Agent text for privacy concerns... don't need harvesting of that info by third parties and it could possibly raise an uproar.

**Comment by Martii** [comment]
@sizzlemctwizzle 
> if I knew the frontend UI was just waiting for me

Btw sorry... I am noisy as I've been told a few times... I just figured out how to read fonts in my distro too so some of my remote issues should fall into place at some point. I understand most of this project *(save for the auth deps in a few areas)*... balancing what is a need vs a want is a continual task and I presume that is my role here as Active Maintainer. I'm trying to not veto/get it done right now as much as possible... but we're all busy elsewhere too.

Like I mentioned earlier is placement of the User-Agent in the UI... then collecting the data in our schemas... then processing it and displaying it.

**Comment by sizzlemctwizzle** [comment]
I just wanted to chime in and say that I hate our current floating comment editor. I'd really rather have it stuck to the bottom or top of the page (if we reversed the order of comments), like GitHub. Maybe the floating could be enabled/disable by the click of a button in the editor and have that setting preserved for the user.

**Comment by Martii** [comment]
> I hate our current floating comment editor

That is changeable... every time jQuery gets updated I have to test to make sure that still works... so sticking it at the tail is an option.

> if we reversed the order of comments

That will affect pagination too... large change.

**Comment by Martii** [comment]
@sizzlemctwizzle 
This is fini as far as I'm concerned... any additional CSS icons will need to be followed up with. I can do that since I scour the site daily if I see any.

If there are any other modifications to this feature please open a new issue ticket.

**Comment by Martii** [comment]
Misc note:

Seriously? XBox 360 is IE 9. Remind me never to get one of these again. They could have at least put 11 on it to match Win 7 time period wise. :\
